lib/recart-nacelle-plugin.js

```diff
@@ -1,7 +1,34 @@
 const getMetafieldsForCheckout = require('./getMetafieldsForCheckout')
+const { saveCart } = require('./cart')
 
-const initRecartPlugin = async (ctx, inject) => {
+const delay = async milliseconds => {
+  return new Promise(resolve =>
+    setTimeout(() => resolve(true), milliseconds)
+  )
+}
+
+const initRecartPlugin = async (ctx) => {
   console.log('Recart Nacelle plugin started')
+
+  let timoutCounter = 0
+  while (!window._recart || !window._recart.isReady()) {
+    if (timoutCounter > 10) throw new Error('Recart is not initialized.')
+    await delay(1000)
```

Comment:
I'd make the delay smaller. Waiting for a whole second would make the loading pretty slow.

How about `200` millisec delay and 50 retries?